packages/frontend-2/components/settings/user/Profile.vue

```diff
@@ -0,0 +1,17 @@
+<template>
+  <div class="md:max-w-xl md:mx-auto">
+    <SettingsSectionHeader title="Profile" text="Manage your profile information" />
+    <template v-if="user">
+      <SettingsUserProfileDetails :user="user" />
+      <hr class="my-6 md:my-10" />
+      <SettingsUserProfileChangePassword :user="user" />
+      <hr class="my-6 md:my-10" />
+      <SettingsUserProfileDeleteAccount :user="user" />
+    </template>
+  </div>
+</template>
+<script setup lang="ts">
+import { useActiveUser } from '~/lib/auth/composables/activeUser'
+
+const { activeUser: user } = useActiveUser()
```

Comment:
Doesn't make sense for this component to manage this - what if there is no user (as in - the user is logged out)? Then you have a weird state where you have the component just with the header and nothing else inside of it.

Pass the user in through a prop and have the parent component validate that this component can even be rendered

Reply:
I refactored these component to all be prop based, with the prop coming from the settings dialog. I added a check in the settings dialog to check if there is an activeUser, if not I close it (as it will never make sense to open it if you are not logged in)